### PR TITLE
enh/BA-133 payrexx subscription

### PR DESCRIPTION
**Pull request by michael-scheurer**

### CODE REVIEW

**Review by elias-summermatter — APPROVED** [review]

**Review by elias-summermatter — CHANGES_REQUESTED** [review]
Tests are failing

File: libs/membership/website/src/lib/subscribe/subscribe-container.tsx [hunk 1]
```diff
@@ -18,6 +18,7 @@ import {produce} from 'immer'
 import {StripeElement, StripePayment} from '@wepublish/payment/website'
 import {useEffect, useMemo, useState} from 'react'
 import {OptionalKeysOf} from 'type-fest'
+import {useRouter} from 'next/router'
```
Comment (Itrulia) [inline libs/membership/website/src/lib/subscribe/subscribe-container.tsx:21]: Can't access framework specific things inside website builder.

File: libs/membership/website/src/lib/invoice-list/invoice-list-item.tsx [hunk 2]
```diff
@@ -153,9 +154,15 @@ export function InvoiceListItem({
...
         {/* @TODO: Remove when all 'payrexx subscriptions' subscriptions have been migrated  */}
         {showPayrexxSubscriptionWarning && (
-          <Alert severity="warning">
-            Wir haben vor einiger Zeit das Membersystem umgestellt. Du verfügst noch über eine alte
-            Membership, die automatisch abgebucht wird.
+          <Alert
+            severity="warning"
+            action={
+              <Link
+                href={`/mitmachen?memberPlanBySlug=${subscription?.memberPlan.slug}&upsell=true&cancelSubscriptionId=${subscription?.id}`}>
```
Comment (Itrulia) [inline libs/membership/website/src/lib/invoice-list/invoice-list-item.tsx:161]: This URL is bajour specific only.

File: libs/membership/website/src/lib/subscribe/subscribe-container.tsx [hunk 3]
```diff
@@ -19,13 +19,20 @@ import {StripeElement, StripePayment} from '@wepublish/payment/website'
...
+ * Payrexx Subscription. Other use cases are possible.
+ */
 export type SubscribeContainerProps<
   T extends OptionalKeysOf<RegisterMutationVariables> = OptionalKeysOf<RegisterMutationVariables>
 > = BuilderContainerProps &
   Pick<BuilderSubscribeProps<T>, 'fields' | 'schema' | 'defaults' | 'extraMoneyOffset'> & {
     successURL: string
     failureURL: string
     filter?: (memberPlans: MemberPlan[]) => MemberPlan[]
+    deactivateSubscriptionId?: string
```
Comment (michael-scheurer) [inline libs/membership/website/src/lib/subscribe/subscribe-container.tsx:35]: See comment above what this deactivateSubscriptionId does. It's basically a api functionality and very useful in this use case for Bajour but is also used for other use cases at Hauptstadt.

File: libs/membership/website/src/lib/invoice-list/invoice-list-item.tsx [hunk 4]
```diff
@@ -152,10 +152,17 @@ export function InvoiceListItem({
...
+        {/* @TODO: Make href used in <Link> component customizable if necessary. You may want to make the custom Bajour filters default filters. */}
         {showPayrexxSubscriptionWarning && (
-          <Alert severity="warning">
-            Wir haben vor einiger Zeit das Membersystem umgestellt. Du verfügst noch über eine alte
-            Membership, die automatisch abgebucht wird.
+          <Alert
+            severity="warning"
+            action={
+              <Link
+                href={`/mitmachen?memberPlanBySlug=${subscription?.memberPlan.slug}&upsell=true&deactivateSubscriptionId=${subscription?.id}`}>
```
Comment (michael-scheurer) [inline libs/membership/website/src/lib/invoice-list/invoice-list-item.tsx:161]: Read comment above in the code. I know this is a Bajour specific url for now. We can change it if needed. Params wouldn't hurt anyway. Thinking about moving those filter params to standard functionality of Website Builder. Use cases would be the same for other media.

File: libs/membership/website/src/lib/subscribe/subscribe-container.tsx [hunk 5]
```diff
@@ -143,6 +148,7 @@ export const SubscribeContainer = <T extends OptionalKeysOf<RegisterMutationVari
             }
           })
         }}
+        cancelSubscriptionId={cancelSubscriptionId as string | undefined}
```
[the remark this answers is not in the document]
Comment (michael-scheurer) [reply libs/membership/website/src/lib/subscribe/subscribe-container.tsx]: See comments in the code now as well as the renaming.

**Review by elias-summermatter — APPROVED** [review]

**Pull request merged**